x-pack/metricbeat/module/azure/monitor/client_helper.go

```diff
@@ -23,9 +27,37 @@ func mapMetrics(client *azure.Client, resources []*armresources.GenericResourceE
 	for _, resource := range resources {
 		for _, metric := range resourceConfig.Metrics {
 			// get all metrics supported by the namespace provided
-			metricDefinitions, err := client.AzureMonitorService.GetMetricDefinitions(*resource.ID, metric.Namespace)
-			if err != nil {
-				return nil, fmt.Errorf("no metric definitions were found for resource %s and namespace %s %w", *resource.ID, metric.Namespace, err)
+			var metricDefinitions armmonitor.MetricDefinitionCollection
+			var err error
+
+			for {
+				metricDefinitions, err = client.AzureMonitorService.GetMetricDefinitions(*resource.ID, metric.Namespace)
+				if err != nil {
+					errorMsg := "no metric definitions were found for resource " + *resource.ID + " and namespace " + metric.Namespace
+
+					var respError *azcore.ResponseError
+					ok := errors.As(err, &respError)
+					if !ok {
+						return nil, fmt.Errorf("%s, %w", errorMsg, err)
+					}
+
+					// Check if the error has the header Retry After.
+					// If it is present, then we should try to make this request again.
+					retryAfter := respError.RawResponse.Header.Get("Retry-After")
```

Comment:
Just to be on safe side does it worth to check respError.ErrorCode==429? 

Wondering if you should also include the 5xx errors as mentioned https://learn.microsoft.com/en-us/azure/architecture/best-practices/retry-service-specific#retry-usage-guidance, because you opened the PR

Reply:
+1 on checking the error code

Reply:
I will add the error code for 429, will leave the 5xx for later enhancement